### PR TITLE
Use golden color to highlight cells that are sometimes possible to edit

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6845

image

image

### CODE REVIEW

**Review by Districh-ru — APPROVED** [review]
Hi, @ihhub, LGTM. Now it will be easier for the map maker to find the action tile if objects' passability is hidden.
I left one optional suggestion. Could you please take a look if it is valuable.

**Comment by ihhub** [comment]
Hi @Districh-ru , I see no comments for your suggestion here :)

File: src/fheroes2/editor/editor_interface.cpp [hunk 1]
```diff
@@ -655,12 +655,16 @@ namespace Interface
                 if ( brushSize.width > 0 && brushSize.height > 0 ) {
                     const fheroes2::Point indices = getBrushAreaIndicies( brushSize, _tileUnderCursor );
 
-                    _gameArea.renderTileAreaSelect( display, indices.x, indices.y );
+                    assert( Maps::isValidAbsIndex( indices.x ) );
+                    const bool isActionObject = MP2::isOffGameActionObject( world.GetTiles( indices.x ).GetObject() );
+                    const bool isRenderForActionObject = ( isActionObject && brushSize.width == 1 && brushSize.height == 1 && _editorPanel.isDetailEdit() );
```
Comment (Districh-ru) [inline src/fheroes2/editor/editor_interface.cpp:660]: @ihhub, ow it was lost ...
I suggested to check `_editorPanel.isDetailEdit()` first and only then call all other functions:
```cpp
const bool isRenderForActionObject = ( _editorPanel.isDetailEdit() && brushSize.width == 1 && brushSize.height == 1 && MP2::isOffGameActionObject( world.GetTiles( indices.x ).GetObject() ) );
```

Comment (ihhub) [reply src/fheroes2/editor/editor_interface.cpp:660]: Sure, we can do this.

**Review by Branikolog — APPROVED** [review]

**Pull request merged**